index.html

```diff
@@ -3456,6 +3456,44 @@ <h2>
         <h2>
 Metadata Structure
         </h2>
+
+        <p>
+Input and output metadata is often involved during the <a>DID Resolution</a>,
+<a>DID URL Dereferencing</a>, and other DID-related processes. The structure
+used to communicate this metadata MUST be a <a data-cite="INFRA#maps">map</a> of
+properties. Each property name MUST be a <a data-cite="INFRA#string">string</a>.
+Each property value SHOULD be a <a data-cite="INFRA#string">string</a>. In the
+rare cases where a non-string value is required, the value and all non-string
+sub-values MUST be expressed as a <a data-cite="INFRA#maps">map</a>, <a
+data-cite="INFRA#lists">list</a>, <a data-cite="INFRA#boolean">boolean</a>,
+number, or  <a data-cite="INFRA#null">null</a> and it MUST be possible to
+losslessly represent the value across all supported representations as
+described in <a href="#core-representations"></a>.
```

Comment:
```suggestion
Input and output metadata is often involved during the <a>DID Resolution</a>,
<a>DID URL Dereferencing</a>, and other DID-related processes. The structure
used to communicate this metadata MUST be a <a data-cite="INFRA#maps">map</a> of
properties. Each property name MUST be a <a data-cite="INFRA#string">string</a>.
Each property value MUST be a <a data-cite="INFRA#string">string</a>,
<a data-cite="INFRA#maps">map</a>, <a data-cite="INFRA#lists">list</a>,
<a data-cite="INFRA#boolean">boolean</a>, or  <a data-cite="INFRA#null">null</a>.
The values within any complex data structures such as maps and lists 
MUST be one of these data types as well. 
All metadata property definitions MUST define the value type, including any additional
formats or restrictions to that value (for example, a string formatted as a date).
It is RECOMMENDED that property definitions use strings for values where possible.
```

Reply:
@jricher - First, I'm mostly good with your text above.  But one question: INFRA numbers are noticeability missing from your text.  I expect there to be use cases where you want to express a number, such as a timeout value, in metadata.  Why not allow that?

FYI, I'd be fine with the restriction that numbers be integers that are small enough that they can be represented exactly in all INFRA representations.

Reply:
Infra doesn't have a native representation for numbers, so I left it off the list of allowed elements. A property would need to define a specific way to serialize a numeric value. We could call that out here.

Reply:
I added "or as a decimal integer" to the example of type value restrictions to call out numbers specifically as a missing format.

Reply:
I agree that we need the "or decimal integer" clause, since some metadata will need to contain integers.

Reply:
Definitely agree, and I think we could potentially define a set of common string formats for metadata properties to re-use, but those will probably come into light once we have more actual metadata properties defined.